qubes-rpc/qvm-template-repo-query

```diff
@@ -49,16 +56,38 @@ touch -r "$hashfile" "$repodir/template.repo"
 RET=0
 
 if [ "$1" = "query" ]; then
-    dnf repoquery "${OPTS[@]}" --qf='%{name}|%{epoch}|%{version}|%{release}|%{reponame}|%{downloadsize}|%{buildtime}|%{license}|%{url}|%{summary}|%{description}|' "$SPEC"
+    if $DNF5; then
+        dnf repoquery "${OPTS[@]}" --qf='%{name}|%{epoch}|%{version}|%{release}|%{repoid}|%{downloadsize}|%{buildtime}|%{license}|%{url}|%{summary}|%{description}|\n' "$SPEC"
+    else
+        dnf repoquery "${OPTS[@]}" --qf='%{name}|%{epoch}|%{version}|%{release}|%{repoid}|%{downloadsize}|%{buildtime}|%{license}|%{url}|%{summary}|%{description}|' "$SPEC"
+    fi
     RET="$?"
 elif [ "$1" = "download" ]; then
     # Download/retry algorithm: take mirrors in random order. In this order,
     # try to download from the first one - if download failed but anything was
     # downloaded - retry from the same one. If download failed and nothing was
     # downloaded, go to the next one. The intention is to retry on interrupted
     # connection, but skip mirrors that are not synchronized yet.
-    urls="$(dnf downloadurl "${OPTS[@]}" --url --all-mirrors "$SPEC" | shuf)"
-    readarray -t urls <<<"$urls"
+    declare -a urls=()
+    if $DNF5 && dnf download --help | grep -q allmirrors; then
+        # The smartest case. DNF5 on Fedora 41 with --allmirrors patch
+        urls="$(dnf download "${OPTS[@]}" --url --allmirrors "$SPEC")"
+        readarray -d ' ' -t urls <<<"$urls"
+        urls=( $(shuf -e "${urls[@]}") )
+    elif $DNF5; then
+        # The middle case. DNF5 on Fedora 41 before --allmirror patch
+        # TODO: Phase out after DNF5 --allmirrors patch is released
+        url="$(dnf download "${OPTS[@]}" --url "$SPEC")"
+        urls+=($url)
```

Comment:
Maybe you can simply change this to `urls=($url)`? I don't think there is anything in the `urls` at this point yet?

Reply:
OK. Will do it as well